ivre/parser/iptables.py

```diff
@@ -0,0 +1,81 @@
+#! /usr/bin/env python
+
+# This file is part of IVRE.
+# Copyright 2011 - 2018 Pierre LALET <pierre.lalet@cea.fr>
+#
+# IVRE is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# IVRE is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with IVRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# Maintainers:
+#   - 2018 Francois CHENAIS <francois.chenais@cea.fr>
+#
+
+"""Support for Iptables log from syslog files
+
+
+Feb  4 05:30:11 pi01 kernel: [3240403.495065] IPTABLES/UDP/IN=enxb827eb8f8a4f OUT= MAC=ff:ff:ff:ff:ff:ff:18:62:2c:7e:45:d0:08:00:45:00:00:ec:00:00:40:00:40:11:df:e3 SRC=192.168.0.254 DST=192.168.0.255 LEN=236 TOS=0x00 PREC=0x00 TTL=64 ID=0 DF PROTO=UDP SPT=138 DPT=138 LEN=216
+
+TODO:
+ - the log does not contain the Year so it's necessary to set it using parameter
+   or changing input with new field 'YEAR' or using current year.
+ - The log contains mac addr from source and destination, is this inifo can be
+   added to neo db ?
+
+"""
+
+import datetime
+from builtins import zip
+from ivre.parser import Parser
+
+
+class Iptables(Parser):
+    """Iptables log generator from a syslog file descriptor"""
+
+    def __init__(self, fname, pcap_filter=None):
+        super(Iptables, self).__init__(fname)
+
+
+    @classmethod
+    def parse_line(self, line):
+        line = line.decode().rstrip('\r\n')
+        if not line:
+            return next(self)
+
+        fields={}
+        # Converts the syslog iptables log into hash
+        for i in line[line.find('IN='):].split():
+            try:
+                (k,v)=i.split('=')
+                fields[k]=v
+            except:
+                fields[i]=''
+
+        # Because day of month can be on 1 or 2 char(s)
+        try:
+            fields['start_time'] = datetime.datetime.strptime(line[:15], '%b  %d %H:%M:%S')
+        except:
+            fields['start_time'] = datetime.datetime.strptime(line[:15], '%b %d %H:%M:%S')
+
+        # Rename fields according to flow2db specifications.
+        fields['src'] = fields.pop('SRC')
+        fields['dst'] = fields.pop('DST')
+        fields['proto'] = fields.pop('PROTO').lower()
+        if fields['proto'] in ('udp', 'tcp'):
+            fields['sport'] = int(fields.pop('SPT'))
+            fields['dport'] = int(fields.pop('DPT'))
+
+        # This data are mandatory but undefined in iptables logs, so make my/a choice.
```

Comment:
**is** mandatory ;-)

Reply:
ho! ho! 
What's the best way to process proto without sport or dport ?

Reply:
I don't get your question. You mean when proto is ICMP for example? I **think** we use type and code for sport & dport, but I'm not really sure. Maybe @fmonjalet knows?

Reply:
Yes.

Reply:
Hum, seems working yet without dport and sport ...

Reply:
Proto is used in IGMP example.

Reply:
OK ... I just realize you were talking about the conjugation 🤣 !!